bankbot.lic

```diff
@@ -0,0 +1,232 @@
+      else
+        return true
+      end
+    else
+      return true
+    end
+  end
+
+  def get_currency_type(denomination)
+    case denomination
+    when /Dok|Dokora.*/i
+      currency_type = "Dokora"
+    when /Lir|Lirum.*/i
+      currency_type = "Lirum"
+    when /Kro|Kronar.*/i
+      currency_type = "Kronar"
+    end
+
+    return currency_type
+  end
+
+  def show_balance_sheet(character)
+    closed_accounts = 0;
+
+    @denominations.each do |type|
+      account_balance = UserVars.bot_bank[character][type]
+
+      if account_balance <= 0 || account_balance.nil?
+        closed_accounts = closed_accounts + 1
+        next
+      end
+
+      money_readable = minimize_coins(account_balance)
+
+      denomination_msg = ""
+      money_readable.each do |amount|
+        denomination_msg = denomination_msg + "#{amount} " 
+      end
+
+      fput "whisper #{character} It looks like your current balance is #{denomination_msg} (#{account_balance}) #{type}."
+    end
+
+    if closed_accounts >= 3
+      fput "whisper #{character} You do not have an account with us."
+    end
+  end
+
+end
+
+arg_definitions = [
+  [
+    { name: 'start', regex: /start/i, description: 'Required: prevents accidentally starting up bankbot' },
+    { name: 'announce', regex: /announce/i, optional: true, description: 'If arg is present, bankbot will announce its presence in LNet' },
+    { name: 'greet', regex: /greet/i, optional: true, description: 'If arg is present, bankbot will greet characters after validating (but only once)' }
+  ]
+]
+
+args = parse_args(arg_definitions)
+
+bankbot = Bankbot.new
+validator = CharacterValidator.new(args.announce, false, args.greet)
+
+@last_room_list = []
+loop do
+  line = script.gets?
+  pause 0.05 unless line
+
+  if DRRoom.pcs != @last_room_list
+    (DRRoom.pcs - @last_room_list).each { |name| validator.validate(name) }
+    @last_room_list = DRRoom.pcs
+  end
+
+  case line
+  when /^(.*) whispers, "(help)"$/i
+    character = Regexp.last_match(1)
+
+    bankbot.help(character) if validator.valid?(character)
+  when /^(.*) offers you a tip of (\d+) (\w+).  Type ACCEPT TIP, to accept it or DECLINE TIP to refuse it.$/
+    character = Regexp.last_match(1)
+    amount = Regexp.last_match(2)
+    denomination = Regexp.last_match(3)
+
+    bankbot.deposit(character, amount, denomination) if validator.valid?(character)
+  when /^(.*) whispers, "(with|withdraw) (\d+)\s*(Kro.*|Lir.*|Dok.*)"$/i
+    character = Regexp.last_match(1)
+    amount = Regexp.last_match(3)
+    denomination = Regexp.last_match(4)
+
+    bankbot.withdraw(character, amount, denomination) if validator.valid?(character)
+  when /^(.*) whispers, "(bal|balance)"$/i
+    character = Regexp.last_match(1)
+
+    bankbot.balance(character) if validator.valid?(character)
+  when /^(.*) whispers, ".*"$/i
+    character = Regexp.last_match(1)
+
+    fput "whisper #{character} I'm sorry, I did not understand that command"
+    bankbot.help(character)
```

Comment:
This call (and the whisper above) should check if the character is validated. I think that's a bug in lichbot, too...